File: src/main/java/nz/laspruca/tcplugin/commands/GiveBaton.java
package nz.laspruca.tcplugin.commands;

import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

import java.util.ArrayList;

import static nz.laspruca.tcplugin.Plugin.baton;
import static org.qrl.tcplugin.TCPlugin.*;

import java.util.Objects;

@nz.laspruca.tcplugin.Command(name = "givebaton")
public class GiveBaton implements CommandExecutor {
	@Override
	public boolean onCommand(CommandSender commandSender, Command command, String s, String[] args) {
		if (baton) {
			int argslen = 0;

			for (String str : args) {
				argslen++;
			}

			if (argslen < 1) {
				Objects.requireNonNull(plugin.getServer().getPlayer(commandSender.getName())).getInventory().addItem(makeBonkyBoi());
			} else {
				for (String str : args) {
					if (str.equals("@a")) {
						for (Player player : plugin.getServer().getOnlinePlayers()) {
							player.getInventory().addItem(makeBonkyBoi());
						}
					} else if (str.equals("@p")) {
						Objects.requireNonNull(plugin.getServer().getPlayer(commandSender.getName())).getInventory().addItem(makeBonkyBoi());
					} else {
						try {
							Objects.requireNonNull(plugin.getServer().getPlayer(str)).getInventory().addItem(makeBonkyBoi());
						} catch (Exception ignored) {
							commandSender.sendMessage("Unable to find player " + str);
						}
					}
				}
			}
		}

		commandSender.sendMessage(
				ChatColor.RED.toString() +
						ChatColor.BOLD.toString() +
						"GIVE BATON HAD BEEN DISABLED BY ORDER OF THE 05 COUNCIL");

		return true;
	}

	private ItemStack makeBonkyBoi() {
		ItemStack is = new ItemStack(Material.STICK, 1);

		ItemMeta im = is.getItemMeta();
		ArrayList<String> lore = new ArrayList<String>();

		lore.add("THIS IS YOUR OFFICIAL TECHNOCRAFT POLICE DPT BATON");
		lore.add("HAVE FUN");
		im.setLore(lore);
		im.addEnchant(Enchantment.KNOCKBACK, 1000, true);
		im.setDisplayName(ChatColor.BLUE.toString() + ChatColor.BOLD.toString() + "TCPD Baton");

		is.setItemMeta(im);

		return is;
	}
}

File: src/main/kotlin/nz/laspruca/tcplugin/commands/GiveBaton.kt
package nz.laspruca.tcplugin.commands

import net.md_5.bungee.api.ChatColor.*
import nz.laspruca.tcplugin.Plugin.Companion.baton
import org.bukkit.ChatColor
import org.bukkit.enchantments.Enchantment
import java.util.ArrayList
import org.bukkit.Material
import org.bukkit.command.Command
import org.bukkit.inventory.ItemStack
import org.qrl.tcplugin.TCPlugin.Companion.plugin
import org.bukkit.entity.Player
import org.bukkit.command.CommandSender
import org.bukkit.command.CommandExecutor
import java.lang.Exception


class GiveBaton() : CommandExecutor {
    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<out String>): Boolean {
        if (baton) {
            if (args.isEmpty()) {
                plugin.server.getPlayer(sender.name)!!.inventory
                    .addItem(makeBonkyBoi())
            } else {
                for (str: String in args) {
                    when (str) {
                        "@a" -> {
                            for (player: Player in plugin.server.onlinePlayers) {
                                player.inventory.addItem(makeBonkyBoi())
                            }
                        }
                        "@p" -> {
                            plugin.server.getPlayer(sender.name)?.inventory?.addItem(makeBonkyBoi())
                        }
                        else -> {
                            try {
                                plugin.server.getPlayer(str)?.inventory?.addItem(makeBonkyBoi())
                            } catch (_: Exception) {
                                sender.sendMessage("Unable to find player $str")
                            }
                        }
                    }
                }
            }
        } else {
            sender.sendMessage(
                "${ChatColor.RED}${ChatColor.BOLD} GIVE BATON HAD BEEN DISABLED BY ORDER OF THE 05 COUNCIL"
            )
        }
        return true
    }
}

private fun makeBonkyBoi(): ItemStack {
    val itemStack = ItemStack(Material.STICK, 1)
    itemStack.itemMeta = itemStack.itemMeta?.apply  {
        this.lore = ArrayList<String>().apply {
            add("THIS IS YOUR OFFICIAL TECHNOCRAFT POLICE DPT BATON")
            add("HAVE FUN")
        }
        addEnchant(Enchantment.KNOCKBACK, 1000, true)
        setDisplayName("$BLUE${BOLD}TCPD Batton")
    }

    return itemStack
}
